authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OuraAuthorizationService.kt

```diff
@@ -27,28 +23,34 @@ class OuraAuthorizationService(
         val authConfig = clients.forSourceType(user.sourceType)
         val deregistrationEndpoint = checkNotNull(authConfig.deregistrationEndpoint)
 
-        val revokeURI = UriBuilder.fromUri(deregistrationEndpoint).queryParam("access_token", accessToken).build().toString()
-
-        val credentials = Credentials.basic(
-            checkNotNull(authConfig.clientId),
-            checkNotNull(authConfig.clientSecret),
-        )
-
-        var requestObj = Request.Builder().apply {
-            url(revokeURI)
-            post(FormBody.Builder().build())
-            header("Authorization", credentials)
-            header("Content-Type", "application/x-www-form-urlencoded")
-            header("Accept", "application/json")
-        }.build()
+        val isSuccess = try {
+            val response = httpClient.submitForm {
```

Comment:
I haven't really used `withContext(Dispatchers.IO)` much, but should it be used here as well?

Reply:
Good find, yes that's right.